### ISSUE TITLE
Provide a programmable JS API around function-runner

### ISSUE THREAD

**Issue by nickwesselman** [Shopify/function-runner-javascript#4]
In addition to facilitating execution via `npx`, would be great if this library allowed you to programmatically invoke the function-runner in test scripts, and even automated integration tests.

[Example of such usage](https://github.com/nickwesselman/polyglot-functions/blob/077021d84e465964441eaf590a536e0866afd8b4/app/test-all-functions.js)

A good validation case for this would be adding an integration test to samples in `function-examples`.

**Comment by andrewhassan** [comment]
@nickwesselman Are you thinking of exposing something like this:

```typescript
// This is the result of function-runner
interface Result {
  name: string;
  size: number;
  memoryUsage: number;
  instructions: number;
  logs: string;
  output: OutputType;
}

export function run<InputType, OutputType>(modulePath: string, input: InputType): Result<OutputType> {
  // ...
}
```

**Comment by andrewhassan** [comment]
I don't believe so since we'll be building this directly in `@shopify/function-runner`. We can modify the code so that it doesn't print that.
